lib/mail/matchers/has_sent_mail.rb

```diff
@@ -128,6 +134,12 @@ def matches_on_subject_matcher?(delivery)
         @subject_matcher.match delivery.subject
       end
 
+      def matches_on_attachments?(delivery)
+        return false if delivery.attachments.empty?
+        return true if @attachments.empty?
+        @attachments.all? { |attachment| delivery.attachments.include?(attachment) }
+      end
+
```

Comment:
Can trade the early returns for a boolean expression:
```ruby
delivery.attachments.any? && @attachments.all? { |a| delivery.attachments.include? a }
```
(When `@attachments` is `[]`, `#all?` will always be `true` so we needn't check that separately.)